components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -236,6 +272,8 @@ int ESP8266Interface::disconnect()
         _conn_stat = NSAPI_STATUS_DISCONNECTED;
     }
 
+    _initialized = false;
```

Comment:
Should be done at the begining of the function.

Reply:
Fixed.